AFQ/mask.py

```diff
@@ -0,0 +1,530 @@
+import numpy as np
+
+import nibabel as nib
+from dipy.segment.mask import median_otsu
+
+import AFQ.registration as reg
+import AFQ.utils.volume as auv
+
+
+def check_mask_methods(mask, mask_name=False):
+    '''
+    Helper function
+    Checks if mask is a valid mask.
+    If mask_name is not False, will throw an error stating the method
+    not found and the mask name.
+    '''
+    error = (mask_name is not False)
+    if not hasattr(mask, 'find_path'):
+        if error:
+            raise TypeError(f"find_path method not found in {mask_name}")
+        else:
+            return False
+    elif not hasattr(mask, 'get_mask'):
+        if error:
+            raise TypeError(f"get_mask method not found in {mask_name}")
+        else:
+            return False
+    else:
+        return True
+
+
+def _resample_mask(mask_data, dwi_data, mask_affine, dwi_affine):
+    '''
+    Helper function
+    Resamples mask to dwi if necessary
+    '''
+    mask_type = mask_data.dtype
+    if ((dwi_data is not None)
+        and (dwi_affine is not None)
+            and (dwi_data[..., 0].shape != mask_data.shape)):
+        return np.round(reg.resample(mask_data.astype(float),
+                                     dwi_data[..., 0],
+                                     mask_affine,
+                                     dwi_affine)).astype(mask_type)
+    else:
+        return mask_data
+
+
+class _MaskCombiner(object):
+    '''
+    Helper Object
+    Manages common mask combination operations
+    '''
+
+    def __init__(self, shape, combine):
+        self.combine = combine
+        if self.combine == "or":
+            self.mask = np.zeros(shape, dtype=bool)
+        elif self.combine == "and":
+            self.mask = np.ones(shape, dtype=bool)
+        else:
+            self.combine_illdefined()
+
+    def combine_mask(self, other_mask):
+        if self.combine == "or":
+            self.mask = np.logical_or(self.mask, other_mask)
```

Comment:
I wonder whether instead of implementing this combiner object you could implement the `__or__` and `__and__` operator for these objects. That way you can do something like: `combined = mask1 or mask2` or `combined mask1 and mask2` to express this idea, returning a new mask object as output. Does that do what you want?

Reply:
The main purpose of this class is to reduce the amount of times I have to write:
```
        if self.combine == "or":
            ...
        elif self.combine == "and":
            ...
        else:
            self.combine_illdefined()
```
Ie it determines whether to use 'and' or 'or'. If we implemented __or__, that might reduce the amount of times we have to write out np.logical_or, but we would still have to use the switch statement above. There may be a better way to name these variables.